azure-quantum/azure/quantum/optimization/problem.py

```diff
@@ -98,34 +120,59 @@ def serialize(self) -> str:
                 "terms": [term.to_dict() for term in self.terms]
             }
         }
-
         if len(self.terms_slc) > 0:
             result["cost_function"]["terms_slc"] = [term.to_dict() for term in self.terms_slc]
 
         if self.init_config:
             result["cost_function"]["initial_configuration"] = self.init_config
 
         return json.dumps(result)
-
+    
+    def serialize_to_proto(self) -> list:
+        """Serializes a problem to a list serialized protobuf messages
+        Every problem is built into a series of protobuf messages 
+        each with 1000 terms and added to a list of proto messages.
+        In this version only ising and pubo are supported. The grouped terms will be added in the
+        subsquent PR.
+        Every message in the list is a byte string
+        """
+        proto_messages = []
+        msg_count = 0
+        terms_remaining = len(self.terms)
+        while terms_remaining > 0:   
+            proto_problem = problem_pb2.Problem()
+            cost_function = proto_problem.cost_function
+            metadata = proto_problem.metadata
+            if msg_count == 0:
+                #cost_function.version = self.version
+                cost_function.type = proto_types[self.problem_type]
+                metadata["name"] = self.name
+            # add 1000 terms per proto message
+            if terms_remaining - 1000 > 0: 
+                for i in range (1000):
+                    term = cost_function.terms.add()
+                    term.c = self.terms[i].c
+                    for j in range (len(self.terms[i].ids)):
+                        term.ids.append(self.terms[i].ids[j])
+            else:
+                # add the remaining terms to the last message
+                for i in range(terms_remaining):
+                    term = cost_function.terms.add()
+                    term.c = self.terms[i].c
+                    for j in range (len(self.terms[i].ids)):
+                        term.ids.append(self.terms[i].ids[j])
+            msg_count += 1
+            terms_remaining -= 1000
+            proto_messages.append(proto_problem.SerializeToString())
+        return proto_messages
+    
     @classmethod
-    def deserialize(
+    def deserialize_from_json(
```

Comment:
i think the convention is to make this a static method and just call it `from_json`